### PR TITLE
deploy-test

### LINKED ISSUE #1927: [Feature]: Optimize CubeFS containerized deployments
### Description of Requirement
- The current CubeFS containerized deployment uses a bash script, which has a single function and only supports one-click deployment and cleanup. It does not support cluster topology modification, single service/role upgrade, one-click upgrade and expansion, cluster operation and maintenance management, etc.
 - The project needs to use the Go language for containerized deployment reconstruction, supports containerized deployment of clusters based on topology files, and supports functions such as one-click upgrade and expansion.
### Project Output Requirements
- Functional documentation of the deployment tool, including functions such as one-click deployment, cleaning, capacity expansion, and upgrade
- According to the documentation, the implementation of the deployment tool code is completed, and the functional verification basically passes

### PR DESCRIPTION
**Pull request by ytyuanxi**
**What this PR does / why we need it**:

**Which issue this PR fixes**:

fixes #1927 

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: deploy/cmd/datanode.go [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+	dataNode := &DataNode{}
+	err = json.Unmarshal(data, dataNode)
+	if err != nil {
+		return nil, err
+	}
+	return dataNode, nil
+}
+
+func writeDataNode(listen, prof, id, localIP string, masterAddrs, disks []string) error {
+	// 将DataNode配置写入DataNode.json文件
```
Comment (baijiaruo) [inline deploy/cmd/datanode.go:49]: Do not use Chinese comments

File: deploy/cmd/datanode.go [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+	config, err := readConfig()
+	if err != nil {
+		log.Println(err)
+	}
+	disksInfo := []string{}
+	for _, node := range config.DeployHostsList.DataNode {
+		diskMap := ""
+		for _, info := range node.Disk {
+			diskMap += " -v " + info.Path + ":/cfs" + info.Path
+			//disksInfo = append(disksInfo, "/cfs"+info.Path+":"+info.Size)
```
Comment (baijiaruo) [inline deploy/cmd/datanode.go:96]: can be removed

File: deploy/cmd/docker.go [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+	cmd := exec.Command("ssh", nodeUser+"@"+node,
+		"docker run -d --name "+containerName+
+			" -v "+dataDir+"/disk/"+containerName+"/data:/cfs/data"+
+			" -v "+dataDir+"/bin"+":/cfs/bin:ro"+
+			" -v "+dataDir+"/disk/"+containerName+"/log:/cfs/log"+
+			" -v "+dataDir+"/conf/"+containerName+".json:/cfs/conf/datanode.json"+diskMap+
+			" -v "+dataDir+"/script/start_datanode.sh:/cfs/script/start.sh"+
+			" --restart on-failure --privileged     --network  host  "+ImageName+
+			"  /bin/sh /cfs/script/start.sh ")
+	//fmt.Println(cmd)
```
Comment (baijiaruo) [inline deploy/cmd/docker.go:81]: can be removed

**Review by baijiaruo — DISMISSED** [review]
LGTM

**Review by sejust — DISMISSED** [review]

**Review by baijiaruo — APPROVED** [review]
LGTM

**Review by leonrayang — APPROVED** [review]
LGTM

**Pull request merged**